hs_core/hydroshare/resource.py

```diff
@@ -559,13 +559,20 @@ def create_new_version_resource(ori_res, new_res):
         res_avail_date = new_res.metadata.dates.all().filter(type='available')[0]
         new_res.metadata.create_element('date', type='available', start_date=res_avail_date.start_date, end_date=res_avail_date.end_date)
 
-    # add Relation element to link source and target resources
+    # add or update Relation element to link source and target resources
     if new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier"):
```

Comment:
All resources should have the identifier element of type 'hydroShareIdentifier'. No need for the ```if``` here.